plugins/hls-eval-plugin/test/Eval.hs

```diff
@@ -147,15 +150,25 @@ tests =
         , testCase
             "Prelude has no special treatment, it is imported as stated in the module"
             $ goldenTest "TPrelude.hs"
+        , testCase
+            "Don't panic on {-# UNPACK #-} pragma"
+            $ goldenTest "TUNPACK.hs"
+        , testCase
+            "Can handle eval inside nested comment properly"
+            $ goldenTest "TNested.hs"
         , testCase "Test on last line insert results correctly" $ do
             runSession hlsCommand fullCaps evalPath $
                 liftIO $ do
                     let mdl = "TLastLine.hs"
                     -- Write the test file, to make sure that it has no final line return
-                    writeFile (evalPath </> mdl) $ "module TLastLine where\n\n-- >>> take 3 [1..]"
+                    writeFile (evalPath </> mdl) "module TLastLine where\n\n-- >>> take 3 [1..]"
                     goldenTest mdl
 #if __GLASGOW_HASKELL__ >= 808
```

Comment:
In a similar spirit we could use `ignoreForGhcVersions ` or `knownBrokenForGhcVersions` instead raw cpp conditions

Reply:
Thank you for your suggestions! I've just switched to `knownBrokenForGhcVersions` and implement `knownBrokenOnWindows`. Now CPP disappears from func-test for eval!